sympy/solvers/polysys.py

```diff
@@ -185,6 +194,18 @@ def solve_generic(polys, opt):
     .. [Cox97] D. Cox, J. Little, D. O'Shea, Ideals, Varieties
     and Algorithms, Springer, Second Edition, 1997, pp. 112
 
+    Raises
+    ========
+
+    NotImplementedError
+        If the system is not zero-dimensional. (does not have a finite
+        number of solutions)
+
+    UnsolvableFactorError
+        If ``strict`` is True and at least one polynomial in the
+        Groebner basis does not have all its solutions expressible in
+        radicals. (after it is converted to a univariate polynomial)
```

Comment:
Here it does make sense to talk about "solution components" so I'd say "If strict is true and not all solution components are solvable in radicals" rather than referring to the Groebner basis.

Also it would be good to show an example of this in the examples section.